packages/ui5-task-pwa-enabler/lib/pwaEnabler.js

```diff
@@ -0,0 +1,286 @@
+const mustache = require("mustache");
+const fs = require("graceful-fs");
+const path = require("path");
+const ui5Fs = require("@ui5/fs");
+const {JSDOM} = require("jsdom");
+
+let ws = undefined;
+let rootDir = undefined;
+
+const swStrategies = {
+    offlinePage: "Offline-Page",
+    offlineCopyWithBackup: "Offline-Copy-With-Backup-Page",
+    offlineCopy: "Offline-Copy",
+    preCache: "Cache-First",
+    advancedCache: "Advanced-Caching"
+};
+
+let default_manifest = {
+    "short_name": "MyApp",
+    "name": "My App Title",
+    "description": "Test App for PWA",
+    "icons": [
+        {
+            "src": "/images/icons/192x192.png",
+            "type": "image/png",
+            "sizes": "192x192"
+        },
+        {
+            "src": "/images/icons/512x512.png",
+            "type": "image/png",
+            "sizes": "512x512"
+        }
+    ],
+    "start_url": "/index.html",
+    "scope": "/",
+    "background_color": "#EFF4F9",
+    "theme_color": "#3F5161",
+    "display": "standalone"
+};
+
+/**
+ * UI5 PWA enabler
+ *
+ * @param {Object} parameters Parameters
+ * @param {module:@ui5/fs.DuplexCollection} parameters.workspace DuplexCollection to read and write files
+ * @param {module:@ui5/fs.AbstractReader} parameters.dependencies Reader or Collection to read dependency files
+ * @param {Object} parameters.options Options
+ * @param {string} parameters.options.projectName Project name
+ * @param {string} [parameters.options.projectNamespace] Project namespace if available
+ * @param {Object} [parameters.options.configuration] Task configuration if given in ui5.yaml
+ * @returns {Promise<undefined>} Promise resolving with <code>undefined</code> once data has been written
+ */
+module.exports = async function ({workspace, dependencies, options: {projectName, projectNamespace, configuration}}) {
+    ws = workspace;
+    rootDir = "/resources/" + projectNamespace;
+    if (!configuration) {
+        throw "Configuration is missing";
+    }
+    switch (configuration.serviceWorker.strategy) {
+        case swStrategies.offlinePage:
+            await addOfflinePageSw(configuration.serviceWorker);
+            break;
+        case swStrategies.offlineCopy:
+            await addOfflineCopySw();
+            break;
+        case swStrategies.offlineCopyWithBackup:
+            await addOfflineCopyBackupSw(configuration.serviceWorker);
+            break;
+        case swStrategies.preCache:
+            await addCacheFirstSw(configuration.serviceWorker);
+            break;
+        case swStrategies.advancedCache:
+            await addAdvancedCachingSw(configuration.serviceWorker);
+            break;
+        default:
+            throw "Unknown service worker strategy";
+    }
+    await addServiceworkerRegistration();
+    await addManifest(configuration.manifest);
+    return undefined;
```

Comment:
In such cases you can. just type `return;` (which implicitly returns `undefined`) or even remove the `return` as it is the last statement and the function by default returns `undefined`.